integrations/object_store/src/service/writer.rs

```diff
@@ -0,0 +1,285 @@
+                format!("Expected size {size} but got {actual_size}"),
+            ));
+        }
+
+        let bytes = body.to_bytes();
+        let payload = PutPayload::from(bytes);
+        let mut opts = format_put_options(&self.args)?;
+
+        // Add size metadata for tracking
+        opts.attributes.insert(
+            Attribute::Metadata("content-size".into()),
+            AttributeValue::from(size.to_string()),
+        );
+
+        let result = self
+            .store
+            .put_opts(&self.path, payload, opts)
+            .await
+            .map_err(parse_error)?;
+
+        // Build metadata from put result
+        let mut metadata = Metadata::new(EntryMode::FILE);
+        if let Some(etag) = &result.e_tag {
+            metadata.set_etag(etag);
+        }
+        if let Some(version) = &result.version {
+            metadata.set_version(version);
+        }
+
+        Ok(metadata)
+    }
+
+    // Generate a unique upload ID that we'll use to track this session
+    async fn initiate_part(&self) -> Result<String> {
+        // Start a new multipart upload using object_store
+        let opts = format_put_options(&self.args)?;
+        let multipart_opts = format_multipart_options(opts);
+        let upload = self
+            .store
+            .put_multipart_opts(&self.path, multipart_opts)
+            .await
+            .map_err(parse_error)?;
+
+        // Store the multipart upload for later use
+        let mut guard = self.upload.lock().await;
+        if guard.is_some() {
+            return Err(Error::new(
+                ErrorKind::Unexpected,
+                "Upload already initiated, abort the previous upload first",
+            ));
+        }
+        *guard = Some(upload);
+
+        // object_store does not provide a way to get the upload id, so we use a fixed string
+        // as the upload id. it's ok because the upload id is already tracked inside the upload
+        // object.
+        Ok("".to_string())
+    }
+
+    /// Upload a single part of the multipart upload.
+    /// Part numbers must be sequential starting from 1.
+    /// Returns the ETag and part information for this uploaded part.
+    async fn write_part(
+        &self,
+        _upload_id: &str,
+        part_number: usize,
+        size: u64,
+        body: Buffer,
+    ) -> Result<MultipartPart> {
+        // Validate that actual body size matches expected size
+        let actual_size = body.len() as u64;
+        if actual_size != size {
+            return Err(Error::new(
+                ErrorKind::Unexpected,
+                format!("Expected size {size} but got {actual_size}"),
+            ));
+        }
+
+        // Convert Buffer to PutPayload
+        let bytes = body.to_bytes();
+
+        // Generate a proper ETag for the part based on its content
+        // This follows the pattern used by many object stores where ETags are MD5 hashes
+        let etag = calculate_part_etag(&bytes, part_number);
```

Comment:
We don't need to do this, and it might be incorrect. Let's just return an empty string since it not used anyways.

Reply:
fixed in dd71f57